src/panel/Table.tsx

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { action, computed } from 'mobx';
+import { observer } from 'mobx-react';
+import * as React from 'react';
+import { KeyboardEvent, memo, PureComponent, ReactNode } from 'react';
+import { Badge, css, Hi, Icon, ResizeHandle } from './widgets';
+import './table.scss';
+import { Column, RowGroup, RowItem, TableStore } from './tableStore';
+
+interface TableProps<T, G> {
+    columns: Column<T>[]
+    renderIconName?: (item: T) => string
+    renderGroup: (group: G) => ReactNode
+    renderCell: (column: Column<T>, itemData: T) => ReactNode
+    store: TableStore<T, G>
+}
+@observer export class Table<T, G> extends PureComponent<TableProps<T, G>> {
+    @computed get gridTemplateColumns() {
+        const {columns} = this.props;
+        return `34px ${columns.map((col, i) => `${(i === 0 ? 22 : 0) + col.width.get()}px`).join(' ')} 1fr`;
```

Comment:
>34px [](start = 16, length = 4)

Can put these values in a single place so we can easily modify the look?

Reply:
Modifying the larger look in a single place might have a few interpretations. We can discuss.

For the meanwhile, I've broken down and explained this string to make it less magical.

---
In reply to: [441694379](https://github.com/microsoft/sarif-vscode-extension/pull/282#discussion_r441694379) [](ancestors = 441694379)